File: idea-plugin/src/main/java/org/ifinalframework/jetbrains/plugins/aio/application/annotation/ElementApplication.java
package org.ifinalframework.jetbrains.plugins.aio.application.annotation;


import org.springframework.core.annotation.AliasFor;
import org.springframework.stereotype.Component;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * ElementApplication
 *
 * @author iimik
 * @since 0.0.1
 **/
@Component
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface ElementApplication {
    @AliasFor("components")
    Class<?>[] value() default {};

    @AliasFor("value")
    Class<?>[] components() default {};
}

File: idea-plugin/src/main/kotlin/org/ifinalframework/plugins/jetbrains/aio/application/annotation/ElementApplication.kt
package org.ifinalframework.plugins.jetbrains.aio.application.annotation

import org.springframework.core.annotation.AliasFor
import org.springframework.stereotype.Component
import kotlin.reflect.KClass


/**
 * ElementApplication
 *
 * @author iimik
 * @since 1.6.0
 **/
@Component
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class ElementApplication(
    @get:AliasFor("components") val value: Array<KClass<*>> = [],
    @get:AliasFor("value") val components: Array<KClass<*>> = []
)
